src/Meilisearch/TaskRessource.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Meilisearch
+{
+    /// <summary>
+    /// Information of the regarding a task.
+    /// </summary>
+    public class TaskRessource
+    {
+        public TaskRessource(int uid, string? indexUid, TaskInfoStatus status, TaskInfoType type,
+            Dictionary<string, object> details, Dictionary<string, string> error, string duration, DateTime enqueuedAt,
+            DateTime? startedAt, DateTime? finishedAt)
+        {
+            Uid = uid;
+            IndexUid = indexUid;
+            Status = status;
+            Type = type;
+            Details = details;
+            Error = error;
+            Duration = duration;
+            EnqueuedAt = enqueuedAt;
+            StartedAt = startedAt;
+            FinishedAt = finishedAt;
+        }
+
+        /// <summary>
+        /// The unique sequential identifier of the task.
+        /// </summary>
+        public int Uid { get; }
+
+        /// <summary>
+        /// The unique index identifier.
+        /// </summary>
+        public string? IndexUid { get; }
+
+        /// <summary>
+        /// The status of the task. Possible values are enqueued, processing, succeeded, failed.
+        /// </summary>
+        public TaskInfoStatus Status { get; }
+
+        /// <summary>
+        /// The type of task. Possible values are indexCreation, indexUpdate, indexDeletion,
+        /// documentAdditionOrUpdate, documentDeletion, settingsUpdate.
+        /// </summary>
+        public TaskInfoType Type { get; }
+
+        /// <summary>
+        /// Detailed information on the task payload.
+        /// </summary>
+        public Dictionary<string, object> Details { get; }
```

Comment:
You can use `dynamic` if you don't know anything about the type :)

```suggestion
        public Dictionary<string, dynamic> Details { get; }
```

Reply:
Thanks I didn't know that